internal/cephfs/nodeserver.go

```diff
@@ -133,42 +156,57 @@ func (ns *NodeServer) NodeStageVolume(
 	}
 
 	// It's not, mount now
-	if err = ns.mount(ctx, volOptions, req); err != nil {
+
+	if err = ns.mount(
+		ctx,
+		m,
+		volOptions,
+		fsutil.VolumeID(req.GetVolumeId()),
+		req.GetStagingTargetPath(),
+		req.GetSecrets(),
+		req.GetVolumeCapability(),
+	); err != nil {
 		return nil, err
 	}
 
 	log.DebugLog(ctx, "cephfs: successfully mounted volume %s to %s", volID, stagingTargetPath)
 
+	if err = fsutil.WriteNodeStageMountinfo(volID, &fsutil.NodeStageMountinfo{
```

Comment:
I think cleanup is missing in the current code also but if the write fails we need to umount the volume to make sure no stale resources are left?

Reply:
Isnt this the clean up https://github.com/ceph/ceph-csi/pull/2634/files#diff-091d7d0d011bb3b3c7ef3d64b375bb563052397a9a97ac7adfa4c94169367f6cR449 ?